File: Event.cpp
/**
@file
@date 2017/07/21
@author Minji and hexoul
@brief
*/
#include "Event.h"

Event::Event(double _lat, double _lng, Time _oDate, Time _ambulDate, eventType _type)
	: lat(_lat), lng(_lng), occuredDate(_oDate), ambulDate(_ambulDate), type(_type)
{
}


/**
@brief compare occured time of two events
@details
@param
@return
*/
Time Event::getOccuredDate()
{
	return occuredDate;
}

/**
@brief return type of event
@details
@param
@return eventType
*/
Event::eventType Event::getEventType()
{
	return type;
}

/**
@brief return coordinates
@details
@param
@return pair of latitude, longitude
*/
std::pair<double, double> Event::getCoordinates()
{
	return std::make_pair(lat, lng);
}

/**
@brief set station index and drone index
@details
@param
@return 
*/
void Event::setStationDroneIdx(int _stationIndex, int _droneIndex)
{
	stationIndex = _stationIndex;
	droneIndex = _droneIndex;
}

/**
@brief return pair of station and drone index
@details
@param
@return pair of station index and drone index
*/
std::pair<int, int> Event::getStationDroneIdx()
{
	return std::make_pair(stationIndex, droneIndex);
}

/**
@brief overload operator
@details
@param
@return 
*/
bool Event::operator<(const Event& event)
{
	return (Time::timeComparator(occuredDate, event.occuredDate));
}

File: Event.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DroneTransferSimulator
{
    class Event :IComparable<Event>
    {
        private double lat, lng;
        private Time occuredDate, ambulDate;
        private int stationIndex, droneIndex;
        private eventType type;

        public enum eventType { E_EVENT_OCCURED, E_EVENT_ARRIVAL, E_STATION_ARRIVAL };

        public int CompareTo(Event obj)
        {
            if (Time.timeComparator(this.getOccuredDate(), obj.getOccuredDate())) return 1;
            if (Time.timeComparator(obj.getOccuredDate(), this.getOccuredDate())) return -1;
            return 0;
        }

        public Event(double _lat, double _lng, Time _oDate, Time _ambulDate, eventType _type)
        {
            lat = _lat;
            lng = _lng;
            occuredDate = _oDate;
            ambulDate = _ambulDate;
            type = _type;
        }

        public Time getOccuredDate()
        {
            return occuredDate;
        }

        public eventType getEventType()
        {
            return type;
        }

        public Tuple<double, double> getCoordinates()
        {
            return new Tuple<double, double>(lat, lng);
        }

        public void setStationDroneIdx(int _stationIndex, int _droneIndex)
        {
            stationIndex = _stationIndex;
            droneIndex = _droneIndex;
        }

        public Tuple<int, int> getStationDroneIdx()
        {
            return new Tuple<int, int>(stationIndex, droneIndex);
        }
    }
}
